proposals/62-nft-storage-for-nft-hack.md

```diff
@@ -0,0 +1,184 @@
+# `nft.storage` for [NFTHack](https://nfthack.ethglobal.co/)
+
+Authors: [@alanshaw](https://github.com/alanshaw)
+
+Initial PR: https://github.com/protocol/web3-dev-team/pull/62
+
+<!--
+This template is for a proposal/brief/pitch for a significant project to be undertaken by a Web3 Dev project team.
+The goal of project proposals is to help us decide which work to take on, which things are more valuable than other things.
+-->
+<!--
+A proposal should contain enough detail for others to understand how this project contributes to our team’s mission of product-market fit
+for our unified stack of protocols, what is included in scope of the project, where to get started if a project team were to take this on,
+and any other information relevant for prioritizing this project against others.
+It does not need to describe the work in much detail. Most technical design and planning would take place after a proposal is adopted.
+Good project scope aims for ~3-5 engineers for 1-3 months (though feel free to suggest larger-scoped projects anyway). 
+Projects do not include regular day-to-day maintenance and improvement work, e.g. on testing, tooling, validation, code clarity, refactors for future capability, etc.
+-->
+<!--
+For ease of discussion in PRs, consider breaking lines after every sentence or long phrase.
+-->
+
+This is a proposal for time limited storage of NFT data on IPFS, backed by Filecoin and provided **free** to NFTHack participants during the hackathon (March 19 2021).
+
+* Register `nft.storage` (or other cool domain name) and expose a [pinning service API](https://blog.ipfs.io/2021-02-19-go-ipfs-0-8-0/#remote-pinning-services).
+
+    This will either be a proxy to [Pinata](https://pinata.cloud/) (or other pinning service) or use PL's own [pinbot](https://twitter.com/ipfspin).
+    
+    This gives us space to experiment. Data storage and retrieval is de-risked by the service we pin the data to. It affords us a playground with "_real_" data that we can store on Filecoin.
+
+    Here's some of the AWESOME things we could build and deploy in this space:
+
+    * Create a library that automatically makes deals on Filecoin for the pinned data (use/adapt the dealbot?). This is mentioned in the [AWS Facade proposal](https://github.com/protocol/web3-dev-team/pull/34) and could be re-used there.
+
+    * Implement a [deal batching service](https://github.com/protocol/web3-dev-team/pull/60) prototype for increased successful deal probability.
+
+    * Run a "[Free Retrieval via IPFS](https://github.com/protocol/web3-dev-team/pull/52) Lotus node" that `nft.storage` can make deals with and that will expose the NFT data to IPFS.
```

Comment:
Perhaps related to the next two bullet points, but should we consider also trying to get an MVP service for "CID/Deal querying"? 

I think one pattern that has emerged with NFTs is we have these end users (you or me) who use an app like foundation who will manage the storage of data as a blackbox-> but we run into the "if X was hit by a bus, do I lose my data". 

One answer to that ( which I think @rvagg  touched on for his batching proposal) - was potentially exposing the CID/Deal info so a user could grab that externally (and take over retrieval and payment) - is it worth trying to also prototype something lightweight here?

Reply:
> I think one pattern that has emerged with NFTs is we have these end users (you or me) who use an app like foundation who will manage the storage of data as a blackbox-> but we run into the "if X was hit by a bus, do I lose my data".

I'm not sure I understand what X is in this context. The storage provider? I would assume that the remote pinning service is doing it's best not to lose my data. Just like I assume Dropbox is doing their best to not lose my data. Perhaps this means multiple redundant copies are stored on Filecoin and I can see value in a library what negotiates deals to achieve this (RAID for Filecoin, except D - Disk is instead Deal 😉).

In IPFS this problem is alleviated a little - anyone that has ever accessed your CID will become a host for it (until they next GC) and can pin it themselves to keep the content alive on the network forever. So even if you only pin to a single provider that gets hit by a bus then there's a _chance_ you'll still be able to get your content from someone else in the network.

> One answer to that ( which I think @rvagg touched on for his batching proposal) - was potentially exposing the CID/Deal info so a user could grab that externally (and take over retrieval and payment) - is it worth trying to also prototype something lightweight here?

So, you'll have the CID of the content immediately. I don't think that it would be possible to take over the deal but you could negotiate a deal yourself with a different miner for the same content.

Assuming the content is _currently_ retrievable, there's nothing stopping you from pinning it yourself to a backup pinning service or your own IPFS node.

In the remote pinning service API you can query the status of a pinned CID. I think it would be a good idea to enhance that response to include details of the Filecoin deal it is currently backed up in so that it can still be retrieved even if the pinning service went away.